### PR TITLE
feat: backport text-in-voice support to v13

### LINKED ISSUE #7995: Caching webhooks messages in voice channel (text for voice channel)
### Which package is this bug report for?

discord.js

### Issue description

When I revise the webhook (in my case interaction webhook) message as a gateway event it results in following error
```
/app/node_modules/discord.js/src/structures/Webhook.js:202
     return this.client.channels?.cache.get(d.channel_id)?.messages._add(d, false) ?? d;
                                                                  ^
  TypeError: Cannot read properties of undefined (reading '_add')
     at InteractionWebhook.send (/app/node_modules/discord.js/src/structures/Webhook.js:202:67)
     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### Code sample

_No response_

### Package version

13.7.0

### Node.js version

16

### Operating system

ubuntu

### Priority this issue should have

High (immediate attention needed)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GuildMessages

### I have tested this issue on a development release

_No response_

### LINKED ISSUE #7996: Can't handle text message in voice channel
### Which package is this bug report for?

discord.js

### Issue description

1. Type something in chat of a voice channel
2. Got "Discord sent a typing packet to a GUILD_VOICE channel ...." in err log
3. Send text message in chat of a voice channel
4. Event channel.on messageCreate not triggered

### Code sample

```typescript
client.on('messageCreate', async (msg) => {
 console.log(msg);
})
```

### Package version

13.7.0

### Node.js version

16.15.0

### Operating system

Ubuntu 20.04.4 LTS

### Priority this issue should have

High (immediate attention needed)

### Which partials do you have configured?

Message

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildBans, GuildVoiceStates, GuildPresences, GuildMessages, GuildMessageReactions, DirectMessageReactions, DirectMessageTyping

### I have tested this issue on a development release

_No response_

### LINKED ISSUE #7998: Support Text In Voice
### Which package is the feature request for?

discord.js

### Feature

Yesterday, the stable version of discord became Text In Voice compatible.

### Ideal solution or implementation

Send a message to Text In Voice. etc.

### Alternative solutions or implementations

_No response_

### Other context

_No response_

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Backports #6921 and #8002 to v13

Closes #7998
Closes #7996
Closes #7995

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/structures/interfaces/TextBasedChannel.js [hunk 1]
```diff
@@ -330,6 +330,44 @@ class TextBasedChannel {
...
+   * channel.fetchWebhooks()
+   *   .then(hooks => console.log(`This channel has ${hooks.size} hooks`))
+   *   .catch(console.error);
+   */
+  fetchWebhooks() {
+    return this.guild.channels.fetchWebhooks(this.id);
+  }
+
+  /**
+   * Options used to create a {@link Webhook} in a {@link TextChannel} or a {@link NewsChannel}.
```
Comment (Jiralite) [inline src/structures/interfaces/TextBasedChannel.js:347]: We'd have to add voice channels to this, but I suppose we could generalise it instead.

```suggestion
   * Options used to create a {@link Webhook} in a guild text-based channel.
```

Comment (advaith1) [reply src/structures/interfaces/TextBasedChannel.js:347]: forum channels can have webhooks; are those "text-based"? maybe just say guild channel

Comment (Jiralite) [reply src/structures/interfaces/TextBasedChannel.js:347]: That's not in version 13. When it's added, it can be amended.

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

File: src/structures/VoiceChannel.js [hunk 2]
```diff
@@ -12,6 +14,18 @@ let deprecationEmittedForEditable = false;
  * @extends {BaseGuildVoiceChannel}
```
Comment (SpaceEEC) [inline src/structures/VoiceChannel.js:14]: This is missing a ` * @implements {TextBasedChannel}` so the properties appear in the documentation.
Looks like I forgot this in my PR 😅

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
Just one documentation change.

**Pull request merged**